android-components/components/browser/state/src/main/java/mozilla/components/browser/state/state/TranslationsStatus.kt

```diff
@@ -0,0 +1,19 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.state
+
+/**
+ * Value type that represents the status of a translation within a [SessionState].
+ *
+ * @property isTranslated The page is currently translated.
+ * @property isTranslateProcessing The page is currently attempting a translation.
+ * @property isTranslateProcessing The page is currently attempting a restoration.
+ */
+data class TranslationsStatus(
```

Comment:
Thanks for taking a look! I had a bit of a debate here. I'm going to have something additionally called `TranslationsState` coming from [GeckoView that will be mapped](https://searchfox.org/mozilla-central/rev/11d085b63cf74b35737d9c036be80434883dd3f6/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/TranslationsController.java#793) to something named similarly (Probably `TranslationsState` proper or something like`GeckoTranslationsState`.)

I'm still thinking about it, but I'm thinking it will be its own property here instead of de-serializing its members directly because it has a separate lifecycle. (e.g, `val translationsState: TranslationState = null` instead of each item individually on `TranslationsStatus`.)

Either naming variant or a new one works for me to avoid these names getting confusing. What do you think?

Reply:
More info - lifecycle for the engine provided `TranslationState` will be coming in via a new delegate (probably the next AC patch I do) on [`onTranslationStateChange` ](https://searchfox.org/mozilla-central/rev/11d085b63cf74b35737d9c036be80434883dd3f6/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/TranslationsController.java#659).

Reply:
If I am understanding this correctly, that would mean we would have both `TranslationsState` and `TranslationsStatus` in the `BrowserState`? My initial reaction is that maybe we would want this just all in one because that's just how the other states are - this is a weakly held opinion. I am happy to just say let's leave the changes as they are and we can rethink this when we add more of the translations state and re-examine our assumptions.

Reply:
I'm aiming for just `TranslationsStatus` on the `BrowserState`.

The picture I have right now is:
```
data class TranslationsStatus(
    val isTranslated: Boolean = false,
    val isTranslateProcessing: Boolean = false,
    val isRestoreProcessing: Boolean = false,
    val translationsEngineState: TranslationsState  = null,
)
```
`translationsEngineState` will be refreshed on `onTranslationStateChange` and `TranslationsState` is a container for the Gecko defined `TranslationsState`.

I think it would be easy for us to revisit when I have a `onTranslationStateChange` patch up too to see what makes the most sense.

Reply:
Hmm, `TranslationsEngineState` might be a better name for the mapped`TranslationsState` coming from GeckoView. Anyway, something to think about.

Reply:
Ah, your picture makes sense to me. Now that I see it, I think we should change to `TranslationsState` so that we are still consistent with the other feature states. We could alias the GV's `TranslationsState` as `GeckoTranslationsState`.

Reply:
Perfect, works for me! 

Refactored here: `TranslationsStatus` -> `TranslationsState`

Will plan: GeckoView's `TranslationsState` map to -> `GeckoTranslationsState`